lib/diameter/src/compiler/diameter_dict_util.erl

```diff
@@ -1281,6 +1313,94 @@ dict(Mod) ->
                     [Mod])
     end.
 
+%% inherited_modules/2
+%% Returns list of inherited modules, without returning module in which avp is defined
+inherited_modules(Mod, Inherits) ->
+    lists:filtermap(fun([_, {_, _, M} | _Names]) ->
+                            AM = ?A(M),
+                            case AM of
+                                Mod ->
+                                    false;
+                                _ ->
+                                    {true, AM}
+                            end
+                    end, Inherits).
+
+%% enums_and_avps_from_modules/1
+%% Returns a list of {Mod, Enums, Avps} from modules
+enums_and_avps_from_modules(Mods) ->
+    lists:map(fun(Mod) ->
+                      Enums = enums_from_module(Mod),
+                      Avps = imported_avps_from_module(Mod),
+                      {Mod, Enums, Avps}
+              end, Mods).
```

Comment:
This looks like a list comprehension, or am I missing something?
[{Mod,enums_from_module(Mod),imported_avps_from_module(Mod)} || Mod <- Mods].
Of course, not sure that it actually improves readability in this case ;)